toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -69,6 +71,11 @@ func (s *SystemConfig) IsValid() (err error) {
 	// Validate BootType
 
 	// Validate HostName
+	if !govalidator.IsDNSName(s.Hostname) || strings.Contains(s.Hostname, "_") {
+		if s.Hostname != "" {
+			return fmt.Errorf("invalid [Hostname]: %s", s.Hostname)
+		}
+	}
```

Comment:
Why two separate `if` statements, instead of adding another condition to the first one?

Reply:
Fixed